components/mail/dovecot/files/dovecot.xml

```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0"?>
+<!--
+
+This file and its contents are supplied under the terms of the
+Common Development and Distribution License ("CDDL"). You may
+only use this file in accordance with the terms of the CDDL.
+
+A full copy of the text of the CDDL should have accompanied this
+source. A copy of the CDDL is also available via the Internet at
+http://www.illumos.org/license/CDDL.
+
+Copyright 2016 Josef 'Jeff' Sipek <jeffpc@josefsipek.net>
+
+-->
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+
+<service_bundle type='manifest' name='dovecot'>
+
+<service
+  name='network/imap/dovecot'
+  type='service'
+  version='1'>
+
+	<dependency
+	  name='filesystem_minimal'
+	  grouping='require_all'
+	  restart_on='none'
+	  type='service'>
+		<service_fmri value='svc:/system/filesystem/local' />
+	</dependency>
+
+	<dependency
+		name='loopback'
+		grouping='require_any'
+		restart_on='error'
+		type='service'>
+		<service_fmri value='svc:/network/loopback' />
+	</dependency>
+
+	<dependency
+		name='network'
+		grouping='optional_all'
+		restart_on='error'
+		type='service'>
+		<service_fmri value='svc:/milestone/network' />
+	</dependency>
+
+	<exec_method
+		type='method'
+		name='stop'
+		exec=':kill'
+		timeout_seconds='60' />
+
+	<instance name='default' enabled='false' >
+
+		<exec_method
+			type='method'
+			name='start'
+			exec='/lib/svc/method/dovecot %m %i'
+			timeout_seconds='60' >
+			<method_context>
+				<method_credential
+					user='root'
```

Comment:
On Thu, Sep 08, 2016 at 20:48:54 -0700, Alexander Pyhalov wrote:
> > +	<exec_method
> > +		type='method'
> > +		name='stop'
> > +		exec=':kill'
> > +		timeout_seconds='60' />
> > +
> > +	<instance name='default' enabled='false' >
> > +
> > +		<exec_method
> > +			type='method'
> > +			name='start'
> > +			exec='/lib/svc/method/dovecot %m %i'
> > +			timeout_seconds='60' >
> > +			<method_context>
> > +				<method_credential
> > +					user='root'
> 
> Why? Allocate UID/GID...

(I already allocated 2 UIDs and a GID, see dovecot.p5m.)

Dovecot uses a multi-process scheme.  The master process forks & execs
helpers (which are currently in /usr/libexec/dovecot/).  These helpers get
various uid/gid based on what they do.  E.g.,

(1) the imap-login process gets minimum dovenull since it deals with
unauthenticated connections.

(2) the imap process gets a uid/gid based on the config - which may be the
uid/gid for the logged in user (e.g., jeffpc == uid 1000), or a globally
used uid when using virtual hosting.

Similar rules apply to the other helpers.

To do this uid changing, the master process needs to start with sufficient
privileges.  Without doing fancy RBAC & privileges(5) stuff, it means
starting as root.

Dovecot is extremely configurable, so I opted for the SMF manifest being as
generous as possible.  (Which is also the easiest manifest to make.)

-- 
Real Programmers consider "what you see is what you get" to be just as bad a
concept in Text Editors as it is in women. No, the Real Programmer wants a
"you asked for it, you got it" text editor -- complicated, cryptic,
powerful, unforgiving, dangerous.